From 4a352e45cc773eb771f76eab68325c886310d5b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Uwe=20Kleine-K=C3=B6nig=20=28The=20Capable=20Hub=29?=
 <u.kleine-koenig@baylibre.com>
Date: Fri, 10 Jul 2026 10:24:12 +0200
Subject: [PATCH 05397/15333] MIPS: ip22-gio: Drop #include of
 <linux/mod_devicetable.h>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

That header doesn't use any device id provided by
<linux/mod_devicetable.h>. Also the users of <asm/gio_device.h> (i.e.
arch/mips/sgi-ip22/ip22-gio.c and drivers/video/console/newport_con.c)
don't use any of them and they compile fine in a mips allmodconfig.

So drop the unused include that only yields unnecessary recompilations
when a device id struct changes.

Signed-off-by: Uwe Kleine-König (The Capable Hub) <u.kleine-koenig@baylibre.com>
Signed-off-by: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
---
 arch/mips/include/asm/gio_device.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/include/asm/gio_device.h b/arch/mips/include/asm/gio_device.h
index f9c102f038d6..6e9d2897347d 100644
--- a/arch/mips/include/asm/gio_device.h
+++ b/arch/mips/include/asm/gio_device.h
@@ -1,6 +1,5 @@
 /* SPDX-License-Identifier: GPL-2.0 */
 #include <linux/device.h>
-#include <linux/mod_devicetable.h>
 
 struct gio_device_id {
 	__u8 id;
-- 
2.55.0

